src/platform/user/authentication/utilities.js

```diff
@@ -27,13 +32,13 @@ const loginUrl = policy => {
 
 function redirect(redirectUrl, clickedEvent, openedEvent) {
   // Keep track of the URL to return to after auth operation.
-  sessionStorage.setItem('authReturnUrl', window.location);
+  sessionStorage.setItem(authnSettings.RETURN_URL, window.location.pathname);
 
   recordEvent({ event: clickedEvent });
 
   return apiRequest(
     redirectUrl,
-    null,
+    { headers: { Accept: 'application/json' } },
```

Comment:
With this, I can merge the BE changes for accommodating both JSON and redirect responses for `sessions#new` requests. And then subsequently can have the FE start using the redirects.